ethstorage/p2p/protocol/syncclient.go

```diff
@@ -1260,6 +1267,40 @@ func (s *SyncClient) ReportPeerSummary() {
 	}
 }
 
+func (s *SyncClient) PurgeBadPeers(closePeerFunc func(peer.ID) error) {
```

Comment:
Why the peer cannot be closed by `RemovePeer()`? And does this operation need a metrics update?

Reply:
RemovePeer is the function that removes a peer from sync client, and closePeerFunc will close the peer connection, and the RemovePeer() will be called after the connection closed.